backend_kvm.go

```diff
@@ -102,6 +147,7 @@ func (b kvmBackend) InitrdModules() []string {
 	return []string{"kernel/drivers/char/virtio_console.ko",
 			"kernel/drivers/virtio/virtio.ko",
 			"kernel/drivers/virtio/virtio_pci.ko",
+			"kernel/drivers/virtio/virtio_pci_modern_dev.ko",
```

Comment:
this module isnt included in debian kernel packages..

Reply:
Fwiw the module was introduced with a while back - April 2015 with Linux v4.0. Perhaps we should mark this module as "optional" so we don't error out, when it's missing?

Reply:
lots of features are introduced in the kernel, doesn't mean that distros have to  build all of it though ;-)

This is what makes me think this series hasn't been tested properly, since this change would outright fail under debian

Reply:
Guilty - I only tried Arch, not Debian. Should we add some documentation what testing "properly" entails? Perhaps even add few CI pipelines - a selection from debos-recipes?

Reply:
If modules listed in array are not available in kernel then it will be ignored for copying. This check is done by scanning modules.builtin  to check if listed driver modules are available or not.
https://github.com/go-debos/fakemachine/blob/9e6ee78d073ec35da1e0c8283c128744f1164386/machine.go#L412

Reply:
that is untrue - if the module isn't built then it won't be present in `builtinModules` and an error will be returned.

for a debian machine:
```
$ ls -la /lib/modules/5.10.0-6-amd64/kernel/drivers/virtio/
total 304
drwxr-xr-x  2 root root  4096 Apr 18 07:39 .
drwxr-xr-x 79 root root  4096 Apr 18 07:39 ..
-rw-r--r--  1 root root 42339 Apr  9 19:17 virtio_balloon.ko
-rw-r--r--  1 root root  7227 Apr  9 19:17 virtio_dma_buf.ko
-rw-r--r--  1 root root 22139 Apr  9 19:17 virtio_input.ko
-rw-r--r--  1 root root 28115 Apr  9 19:17 virtio.ko
-rw-r--r--  1 root root 54107 Apr  9 19:17 virtio_mem.ko
-rw-r--r--  1 root root 22115 Apr  9 19:17 virtio_mmio.ko
-rw-r--r--  1 root root 57379 Apr  9 19:17 virtio_pci.ko
-rw-r--r--  1 root root 52403 Apr  9 19:17 virtio_ring.ko


$ cat /lib/modules/5.10.0-6-amd64/modules.builtin | grep virtio | wc -l
0

```

Reply:
Actually having a closer look, I think we can make this much cleaner and better by parsing `modinfo` output (or using a golang library to that affect):
 - use the `filename` token - we can drop the a) hard-coded kernel module path b) the `modules.builtin` parsing and c) the compression detection/fallback code
 - use the `depends` token and error out if module is missing - we add each dependency, in the above case `virtio_pci_modern_dev` is a dependency of `virtio_pci`

Potentially for a future MR:
Add any `softdep` and `firmware` entries but do not error out, if copying fails,